src/User/AvatarUploader.php

```diff
@@ -46,7 +47,7 @@ public function upload(User $user, Image $image)
 
     public function remove(User $user)
     {
-        $avatarPath = $user->avatar_path;
+        $avatarPath = Arr::get($user->getAttributes(), 'avatar_url');
```

Comment:
This breaks encapsulation of our model object. :confused: 

Instead, I think I would prefer to see a change to the `AvatarChanged` event to hold the following four values:
- the user whose avatar was changed
- the avatar
- the old `avatar_url` value
- the new `avatar_url` value

...and then implement avatar deletion in an event listener.

Thoughts?

Reply:
Because AvatarUploader.php is at the same encapsulation level, which means AvatarUploader can be seen as at the same package level with User.php, so I regard it's acceptable to call `Arr::get($user->getAttributes(), 'avatar_url');` in AvatarUploader.php.

Although, it also makes sense to put both old avatar_url and new avatar_url into AvatarChanged, and remove old avatar with listening to AvatarChanged event.

I suggest to accept this pull request first, because it's a bug rather than an enhancement. And make a new issue to make the enhancement. I'll do that issue then.